### PR TITLE
feat(BUX-750): add admin contact endpoints

### PR DESCRIPTION
**Pull request by pawellewandowski98**
# Pull Request Checklist

- [ ] 📖 I created my PR using provided  : [CODE_STANDARDS](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_STANDARDS.md)
- [ ] 📖 I have read the short Code of Conduct: [CODE_OF_CONDUCT](https://github.com/bitcoin-sv/spv-wallet/blob/main/.github/CODE_OF_CONDUCT.md)
- [ ] 🏠 I tested my changes locally.
- [ ] ✅ I have provided tests for my changes.
- [ ] 📝 I have used conventional commits.
- [ ] 📗 I have updated any related documentation.
- [ ] 💾 PR was issued based on the Github or Jira issue.

# Description

Added contacts endpoints for admin:
- POST   `/admin/contact/search` - search for contacts
- PATCH `/admin/contact/:id` - update contact name and metadata
- PATCH `/admin/contact/rejected/:id` - reject contact 
- PATCH `/admin/contact/unconfirmed/:id`- change confirmed contact to unconfirmed

### CODE REVIEW

File: models/filter/contact_filter.go [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+	// Column names come from the database model, see: /engine/model_contact.go
+	applyIfNotNil(conditions, "id", d.ID)
+	applyIfNotNil(conditions, "full_name", d.FullName)
+	applyIfNotNil(conditions, "paymail", d.Paymail)
+	applyIfNotNil(conditions, "pub_key", d.PubKey)
+	applyIfNotNil(conditions, "status", d.Status)
+
+	return conditions
+}
```
Comment (chris-4chain) [inline models/filter/contact_filter.go:25]: The filters are introduced for `Transactions` and `Destinations` for now. 
Besides, I can see that, the `ContactFilter` is not used, so I would recommend to remove it.

File: actions/admin/contact.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+		return
+	}
+
+	contact, err := a.Services.SpvWalletEngine.UpdateContact(
+		c.Request.Context(),
+		id,
+		reqParams.FullName,
+		&reqParams.Metadata,
+	)
+	if err != nil {
```
Comment (arkadiuszos4chain) [inline actions/admin/contact.go:83]: Handle different errors:

```

	if err != nil {
		switch {
		case errors.Is(err, engine.ErrContactNotFound):
			c.JSON(http.StatusNotFound, err.Error())
		case errors.Is(err, engine.ErrContactIncorrectStatus):
			c.JSON(http.StatusUnprocessableEntity, err.Error())
		default:
			c.JSON(http.StatusInternalServerError, err.Error())
		}
		return
	}

**Review by arkadiuszos4chain — CHANGES_REQUESTED** [review]

**Comment by arkadiuszos4chain** [comment]
I forgot to include two additional endpoints in the ticket. Please implement the following:

`DELETE admin/contact/{id}`
`PATCH admin/contact/accepted/{id}`

My apologies

File: models/filter/contact_filter.go (hunk 1 above)
Comment (pawellewandowski98) [reply models/filter/contact_filter.go:25]: `ContactFilter` is used in /admin/contacts/search endpoint

File: actions/admin/contact.go [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+		reqParams.Metadata,
+		reqParams.Conditions.ToDbConditions(),
+		reqParams.QueryParams,
+	)
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	contracts := make([]*models.Contact, 0)
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply actions/admin/contact.go]: Done

File: actions/admin/contact.go [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+// @Router		/v1/admin/contact/{id} [patch]
+// @Security	x-auth-xpub
+func (a *Action) contactsUpdate(c *gin.Context) {
+	var reqParams UpdateContact
+	if err := c.Bind(&reqParams); err != nil {
+		c.JSON(http.StatusBadRequest, err.Error())
+		return
+	}
+
+	id := c.Param("id")
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply actions/admin/contact.go:70]: Done

File: engine/action_contact.go [hunk 5]
```diff
@@ -103,15 +104,65 @@ func (c *Client) AddContactRequest(ctx context.Context, fullName, paymailAdress,
...
+
+	if err = contact.Save(ctx); err != nil {
+		return nil, fmt.Errorf("updating %s contact failed. Reason: %w", id, err)
+	}
+
+	return contact, nil
+}
+
+func (c *Client) AdminChangeContactStatus(ctx context.Context, id string, status ContactStatus) (*Contact, error) {
+	contact, err := getContactByID(ctx, id, c.DefaultModelOptions()...)
```
[the remark this answers is not in the document]
Comment (pawellewandowski98) [reply engine/action_contact.go:146]: Done

File: actions/admin/contact.go (hunk 2 above)
Comment (pawellewandowski98) [reply actions/admin/contact.go:79]: Done, prepared method to check it and it is used in all contact endpoints where different errors can be returned

**Review by arkadiuszos4chain — APPROVED** [review]

File: models/filter/contact_filter.go (hunk 1 above)
Comment (chris-4chain) [reply models/filter/contact_filter.go:25]: If it works, let's leave it 👌 less work for me 👍

**Review by christomecki — APPROVED** [review]

**Review by chris-4chain — APPROVED** [review]

**Pull request merged**